examples/jetstream/api-migration/deno/main.ts

```diff
@@ -0,0 +1,207 @@
+// (where the client can be overwhelmed with messages). However, this
+// API is hard to use:
+opts = consumerOpts()
+  .ackExplicit()
+  .manualAck();
+
+// pullSubscribe() would create a subscription to process messages
+// received from the stream, but would require a `pull()` to trigger
+// a request on the server to yield messages
+const pullSub = await js.pullSubscribe("events.>", opts);
+const done = (async () => {
+  for await (const m of pullSub) {
+    console.log(`legacy pull subscriber received ${m.subject}`);
+    m.ack();
+    if (m.info.pending === 0) {
+      return;
+    }
+  }
+})();
+
+// To get messages flowing, you needed to call `pull()` on
+// the subscription
+pullSub.pull({ batch: 15, no_wait: true });
+// and also do so at some interval to keep it going - however
+// there was no coordination between the processing of the
+// messages and the triggering of the pulls:
+const timer = setInterval(() => {
+  pullSub.pull({ batch: 15, no_wait: true });
+}, 1000);
+
+await done;
+clearInterval(timer);
+
+// New API for consuming messages on a stream, is way more ergonomic
+// and easier to use. Internally it uses a "pull consumer", handles the
+// requesting of messages under the hood. As an user, you simply
+// can specify how many messages you want to buffer, and the library will
+// do its best to keep that up for you.
+//
+// The new API doesn't automatically create or update consumers. This
+// is something that the JetStreamManager API does rather well. Instead,
+// you simply use JetStreamManager to create your consumer:
+await jsm.consumers.add("EVENTS", {
+  name: "my-ephemeral",
+  ack_policy: AckPolicy.Explicit,
+});
+
+// To process messages, you retrieve the consumer, by specifying the name
+// of the stream and the name of the consumer
+const consumerA = await js.consumers.get("EVENTS", "my-ephemeral");
+
+// With a consumer in hand, you can now retrieve messages - in different ways.
+// Firstly, we'll discuss consume, this is analogous to the push consumer example
+// above:
+const messages = await consumerA.consume();
+for await (const m of messages) {
+  console.log(`consume received ${m.subject}`);
+  m.ack();
+  if (m.info.pending === 0) {
+    break;
+  }
+}
+// if you wanted to preempt delete the consumer you can - however
+// this is something you should do only if you know you are not
+// going to need that consumer.
+await consumerA.delete();
+
+
+// Let's create a new consumer, this time a durable
+await jsm.consumers.add("EVENTS", {
+  durable_name: "my-durable",
+  ack_policy: AckPolicy.Explicit,
+});
+
+// The different ways of getting messages from the consumers are there
+// to help you align the buffering requirements of your application with
+// what the client is doing.
+
+// The legacy API provided a way of retrieving a single message:
+const m = await js.pull("EVENTS", "my-durable")
+  .catch((err) => {
+    // possibly no messages found
```

Comment:
Thioe comments in block do not look nicely in rendered example (same applies to other inline comments in this example):

<img width="988" alt="image" src="https://github.com/ConnectEverything/nats-by-example/assets/15184508/004d3024-44ea-4083-aa52-18155de6000c">